crates/primitives/src/cache.rs

```diff
@@ -0,0 +1,33 @@
+use std::num::NonZeroUsize;
+
+use lru::LruCache;
+use sov_rollup_interface::services::da::DaService;
+
+pub struct L1BlockCache<Da>
```

Comment:
I am against that struct in `primitives` crate because it's not used anywhere but in fullnode-runner AND it's used only by_number, not by_hash.

Reply:
Prover is also using the L1 block hash.

But i agree that by_hash is not needed anymore. Which makes #878 irrelevant.